src/language-js/print/ternary.js

```diff
@@ -153,6 +159,102 @@ function printTernaryTest(path, options, print) {
   return printed;
 }
 
+const ancestorNameMap = new Map([
+  ["AssignmentExpression", "right"],
+  ["VariableDeclarator", "init"],
+  ["ReturnStatement", "argument"],
+  ["ThrowStatement", "argument"],
+  ["UnaryExpression", "argument"],
+  ["YieldExpression", "argument"],
+]);
+function shouldExtraIndentForConditionalExpression(path) {
+  const node = path.getValue();
+  if (node.type !== "ConditionalExpression") {
+    return false;
+  }
+
+  const parentName = path.callParent((path) => path.getName());
+  if (!parentName) {
+    return false;
+  }
+
+  let ancestorCount = 1;
+  let ancestor = path.getParentNode(ancestorCount);
+  while (isChainElement(ancestor)) {
+    ancestor = path.getParentNode(ancestorCount++);
+  }
+  if (ancestor.type === "BinaryExpression") {
+    return false;
+  }
+  const checkAncestor = (node) => {
+    const name = ancestorNameMap.get(ancestor.type);
+    return ancestor[name] === node;
+  };
+
+  const parent = path.getParentNode();
+
+  /**
+   * foo = (
+   *   condition
+   *     ? first
+   *     : second
+   * ) as SomeType;
+   */
+  if (parent.type === "TSAsExpression" && checkAncestor(parent)) {
+    return true;
+  }
+
+  /**
+   * foo = (
+   *   condition
+   *     ? first
+   *     : second
+   * )(arguments);
+   */
+  if (
+    (parent.type === "CallExpression" ||
+      parent.type === "OptionalCallExpression") &&
+    checkAncestor(parent) &&
```

Comment:
Can't you reuse `ancestorCount` here instead of climbing up the `path` one more time in `getChainRoot`?

Also what about the case where `parent.type === "TSNonNullExpression"`?

Reply:
I meant to write this under `parent.type === "MemberExpression" || parent.type === "OptionalMemberExpression"`. But maybe you should do the chain root check here too? After all, the search for ancestor above considers calls chain elements.

Reply:
I was able to get rid of the `getChainRoot` function by reusing the `ancestorCount`, thanks!